corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -584,6 +584,7 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
             this.dynamicSearchEnabled = !(options.disableDynamicSearch || this.smallScreenEnabled) &&
                 (toggles.toggleEnabled('DYNAMICALLY_UPDATE_SEARCH_RESULTS') && this.options.sidebarEnabled);
+            this.searchOnClear = (options.searchOnClear && !this.smallScreenEnabled);
```

Comment:
should this check for `this.searchOnClear = (options.searchOnClear && this.dynamicSearchEnabled);` to clarify requirement and add an extra check?

Reply:
no, same reason as this [comment](https://github.com/dimagi/commcare-hq/pull/33983#discussion_r1456332158)

But it is dependent on Split screen case search FF. I haven't quite figured out where the FF checks should happen as opposed to just putting it in every step. Right now, with these steps
1. SSCS enabled and searchOnClear enabled -> FP menu response sets "searchOnClear" as True
2. SSCS is disabled, search_on_clear is still defined as True on module model -> FP menu response still sets "searchOnClear" as True

I'm thinking I should add the FF check on FP side, otherwise menuResponse is wrong. Do you think that makes sense?

Reply:
I decided to do a check before populating the suite [2168b86](https://github.com/dimagi/commcare-hq/pull/33983/commits/2168b86fc9f306bf50a6e00792f8d0797338c0d9). It'll require app rebuild but I think that makes sense as that's how most of our FF operates.

Reply:
agreed on approach of checking before populating the suite to keep the logic within this repo.